instaclustr/data_source_clusters.go

```diff
@@ -0,0 +1,85 @@
+package instaclustr
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceClustersList() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceClustersListRead,
+
+		Schema: map[string]*schema.Schema{
+			"cluster": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cluster_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"cluster_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"cassandra_version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"node_count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"running_node_count": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"derived_status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"sla_tier": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"pci_compliance": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceClustersListRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Client
+	d.SetId("instaclustr-clusters")
+
+	log.Print("[INFO] Listing all active clusters")
+	clusterList, err := client.ListClusters()
+	if err != nil {
+		return fmt.Errorf("[Error] Error reading cluster credentials: #{err}")
+	}
+
+	clusters := make([]map[string]interface{}, 0)
+
+	for _, c := range *clusterList {
+		cluster := make(map[string]interface{})
```

Comment:
Please ignore if too hard - but could we use `mapstructure` to map the response model to the schema?

```
mapstructure.Decode(c, &cluster)
```

might do the trick, provided that the `ClusterListItem` has the required mapstructure metadata (see `ClusterProvider` for example)